### ISSUE TITLE
[Query] Mechs

### ISSUE THREAD

**Issue by Xerma** [tsunamayo/Skywanderers#1526]
Through a long and drawn out discussion in the discord today, numerous people have become very curious about the plans for mechs and how useful they might or might not be in the future. Flight was the most curious of the topics for mechs seeing as that feature for mechs could also pose a problem for gameplay balancing. 

If mechs can fly like ships then there seems to be a balance issue or complexity between mechs in space and ships, which are meant to maneuver in space. Having mechs fly sort of negates a ships superiority in space/air in a way. This might also break ground combat since mechs could just pop up and start fighting ships and then just plop back down to the ground and continue there. Another issue is if mechs can fly in space and so can ships, then why not hovercraft, cars, etc. Each type of entity should be specifically good at something while keeping to their own 'realm'. Such as ships are good in space and air while hovercraft/mechs are good for ground battles, I do recall you saying you'd like for ground battles to be a thing which many of us still hope to see one day as well since most other games neglect this aspect.

As stated in discord by Ultrak2k, a solution might be to just dock a ship to the mech to act as it's transport which would also be a really cool thing to see and need to do. Although flight may not be a good gameplay choice, jetpacks could be a thing for mechs to still maneuver a little more but also have flight time be temporary which would give mechs a uniqueness about them.

Another cool thing would be to have mechs be good on ground but also be great underwater, should that ever be a thing.

All in all, each entity type should specialize in an element like hovercraft are faster on ground than mechs and can act as a transport on the ground (which would make them a cheaper method of moving ground products than with ships depending on distance), mechs could maneuver more easily or have higher base armor, tanks are beefy and slow but strong, cars do whatever they will do and ships navigate and carry these other items to where they need to be.

**Comment by ultrak2k** [comment]
Full support from me

**Comment by Aiwii** [comment]
I agree, dedicated craft for dedicated purposes.

**Comment by ZachZent** [comment]
Tanks roll
Cars drive
Hovercraft hover
Mechs walk
Monorails run
Ships fly

No need to crossbreed

**Comment by ExodistSKY1** [comment]
Short range jet packs for mechs = Yes please 

spaceship style flight for mechs = No thank you

**Comment by Garrett-C** [comment]
Yeah, I wouldn't ever want mechs to work as an alternative to fighters but It would be nice for them to be able to do some stuff in space combat. I disagree with the "No crossbreeding" argument but I don't think mechs should even be close to being an alternative to fighters. The idea of some form of jetpack that mechs could have added is a good one. It could allow for some space capabilities with more of a directional dash than a flight system. This could even be incorporated on the ground giving them the option to quickly dart in a direction.

**Comment by ZachZent** [comment]
That is exactly what it should be. Mechs should, with the proper equipment, be able to maneuver in space like an EVA suit. Would be useful for combat like boarding (gif below) and utility like mining.
https://gfycat.com/shorttermregalladybug

**Comment by tsunamayo** [comment]
Okay interesting debate here.
My question is, what happens if you put thruster on your mech, put a spaceship core, link it to a control and activate the control? Nothing because of the mech core? Or should I prevent puting the spaceship core? 
I think that would be an artificial restriction, and a serious missed oportunity to build a transformer like mech/plane. 
So my planned way forward was simply to integrate mech into the energy system (ie walking require energy), and let the player figure out how he wants to spend his energy/heat. More weapon? Faster walk? Fly ability?
Then it is pretty much a question of control, do we want to control mech and fly like mech in space as opposed to plane like control. I need to put more thinking into this, and have an unified approach with spaceship control.

Thanks for the interesting contribution!

**Comment by ultrak2k** [comment]
> Okay interesting debate here.
> My question is, what happens if you put thruster on your mech, put a spaceship core, link it to a control and activate the control? Nothing because of the mech core? Or should I prevent puting the spaceship core?
> I think that would be an artificial restriction, and a serious missed oportunity to build a transformer like mech/plane.
> So my planned way forward was simply to integrate mech into the energy system (ie walking require energy), and let the player figure out how he wants to spend his energy/heat. More weapon? Faster walk? Fly ability?
> Then it is pretty much a question of control, do we want to control mech and fly like mech in space as opposed to plane like control. I need to put more thinking into this, and have an unified approach with spaceship control.
> 
> Thanks for the interesting contribution!

Even if mechs couldn’t have spaceship cores, you could dock them to a ship that looks like part of the mech, allowing it to fly but not allowing you to walk or shoot with the mech if you’re controlling the ship. People would begin to ask “why can’t my car/tank fly” if mechs can fly, and having total free flying vehicles I believe would upset combat even if they were weaker. A solution like jump jets for mechs in this case gives mechs a useful specialised ability that only they can have, giving an incentive to use them, since people have been arguing if they will be good or not.

**Comment by ZachZent** [comment]
> I think that would be an artificial restriction, and a serious missed oportunity to build a transformer like mech/plane.

Maybe I'm piping up the wrong tree, but the past year of updates feels like it would (inadvertently) counter transformers. Adding collisions, the upcoming physics updates, and the need to reduce child entities. To make a good looking transformer would be very difficult even without collisions. I have a feeling most people will just end up sticking thrusters on to instantly become a ship.
I support mechs being able to have short use jetpacks and/or jump jets on ground and some limited maneuverability (not flight) in space, but in the end allowing mechs to fly like a ship, you will just get people taking advantage of the system. A vehicle that has access to all areas of the game: ground, atmo, space, interiors. Besides, how do you make a mech flying look good rather then looking stiff without spending a year animating and accounting for every type of mech-thruster arrangement players decide to use. Jetpack style, foot thruster style, ironman hand thrusters just to name a few

**Comment by tsunamayo** [comment]
I dont plan on making it look good. If you walk you get a walk anim and thats it. If you fly then no anim. Maybe I could make a airborn stance but thats it.
 
If I forbid it (what you suggest), then I can already hear people asking me to allow this. I mean if somebody want to make an hybrid why should I prevent it? 
I am not concerned about balance, a multi-role build is always gonna be weaker than a specialized one for a given task. ie a flying mech is gonna be weaker than a simple walking mech (for a given size/energy). But it can fly - and potentially escape.

**Comment by Aiwii** [comment]
Well as long as Hybrids aren't better than their specialized counterparts in their roles this seems fine also.

**Comment by ultrak2k** [comment]
Can we atleast have mech jumpjets then lol

**Comment by ExodistSKY1** [comment]
Flight animation stance = Great idea (People are drawn to what looks neat and awesome, if you got a mech flying with no animation looking like a stiff Popsicle... it doesn't look good)

I agree if a multi role mech is weaker than a dedicated mech all is good!

**Comment by Sei-Neko** [comment]
I'm late to the party, but I would like to contribute my ideas to this topic. However, I would like to clear up on some points. When we're talking about mechs 'flying', are we refering to flying under gravity? If so, I think I would be in the same boat to restrict that/making it weaker as a multi-role machine rather than a dedicated machine.
If we're refering to maneuvering in space, though, I believe that's a completely seperate topic from atmospheric flight. 
Thinking logically, what difference is there between a mech and a small fighter in space? A mech would have thrusters spread out around it's body and on limbs, granting it superior maneuverability while having a lower top speed to something like a fighter, whose main thrusters are all orientated in a single direction. In game balance terms, this could be implemented by allowing mechs to fight in full 6DOF with equal movement speed in all directions, while fighters could behave much like modern fighter jets with high top speed and high agility. I think this could satisfy both camps as a skilled fighter pilot would be able to shoot down a mech and vice versa.

**Comment by codeki** [comment]
New here, so I might be missing key points in this discussion, but my primary interest is in building transforming mechs. So far the UI/UX of Starship EVO have been a delight and smooth as butter.  I don't even mind when I have to rebuild a major part to fix a parent/child issue :D  

https://steamcommunity.com/sharedfiles/filedetails/?id=2429985531

So, my current mech model has 3 modes and 2 different yokes. The spaceship yoke is for Fighter and Guardian modes. The transformation to Guardian leads to a natural tradeoff of forward thrust for greater agility, but it's still an aircraft basically. And in order to engage in robot fun, you have to unyoke, transform to Battroid (robot) mode and active the secondary yoke. Currently, this is just a turret connected to the head and hand-held gun, but I could imagine this being a mech computer. 

Seems to me that having to unyoke/reyoke during a transformation isn't that different from the idea of docking a craft into another craft and switch controls. In both cases you are switching controls and computers, right?  Part of the fun of building transforming mechs is the challenge of cramming it all into one vehicle, accepting compromises for performance compared to single-purpose builds.
